### PR TITLE
feat: support for opening pages in new window

### PR DESCRIPTION
**Pull request by hougesen**
Added support for opening pages in a new tab by holding down ctrl when clicking.  

The routeToPage function can be simplified further by turning the outer switch statement into an if statement like this: 
```js
// ./frontend/src/utils.ts
export const routeToPage = (mediaId: string, mediaType: string = undefined, ctrlDown: boolean = false) => {
    let url;
    if (["movie", "tv", "person"].indexOf(mediaType) !== -1) {
        url = `/${mediaType}/${mediaId}`;
    } else {
        switch (mediaId[0]) {
            case "t":
                url = `/tv/${mediaId.slice(1)}`;
                break;
            case "m":
                url = `/movie/${mediaId.slice(1)}`;
                break;
            case "p":
                url = `/person/${mediaId.slice(1)}`;
                break;
            default:
                url = "/";
                break;
        }
    }

    if (ctrlDown) {
        window.open(url);
    } else {
        window.location.href = url;
    }
};
```

### CODE REVIEW

File: frontend/src/utils.ts [hunk 1]
```diff
@@ -30,18 +31,24 @@ export const routeToPage = (mediaId: string, mediaType: string = undefined) => {
...
-                    window.location.href = "/";
+                    url = "/";
             }
     }
+
+    if (ctrlDown) {
+        window.open(url);
+    } else {
+        window.location.href = url;
+    }
```
Comment (AndreasPB) [inline frontend/src/utils.ts:51]: Do you know if there is a native JS/TS way to get this behavior where we wont have to describe every type of click (that also supports middle mouse + shift click)?

**Review by AndreasPB — COMMENTED** [review]
Thanks for the contribution - This is definitely something Streamchaser should do :)

File: frontend/src/utils.ts (hunk 1 above)
Comment (hougesen) [reply frontend/src/utils.ts:51]: You *should* be able to check which mouse button is pressed by checking the button state 
[https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events#determining_button_states](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events#determining_button_states)

The PointerEvent where the ctrlDown boolean comes from also support modifier keys like alt and shift. 

If you are interested in non modifier keys I believe you have to do something like this. While has been said I don't really have that much experience with Svelte, so there might be an easier way to do it. 

```ts
let lastKeyPress: string | null = null;

window.addEventListener("keydown", (e: KeyboardEvent) => {
  lastKeyPress = e?.key;
});

window.addEventListener("keyup", (e: KeyboardEvent) => {
  lastKeyPress = null;
});

```

**Comment by Primdahl26** [comment]
Hey thanks so much for the PR! I agree that it gets annoying not being able to open stuff in new tabs on streamchaser! 🤣 

With that said i think it might be silly to write javascript handling for all the different key handlers, when if we use `<a href="">` we get them all for free. 

If we look at our index page's [media component](https://github.com/streamchaser/streamchaser/blob/master/frontend/src/components/media_search.svelte), we build a figure containing the figure as below:
```
{#each media.hits as media, mediaIndex}
    <div
        on:click={routeToPage(media.id)}
        <figure>
            <img
                src="{LOW_RES_IMG_URL}{media.poster_path}"
                alt="media poster"
            />
        </figure>
...
```
If this was made using `<a href>` i believe this would be the optimal solution, since this also lets crawlers browse the items. This would also have to be done for images in the different detailsviews aswell.

We are all mid exam stress ATM, but will prio to get this done ASAP. Feel free to hit a comment if you disagree about this solution. And also of course of you wish to try it out that would be awesome! 😎

**Comment by Primdahl26** [comment]
Closing since this is done in #122

**Pull request closed without merging**